cardano-node/src/Cardano/CLI/Tx/Generation.hs

```diff
@@ -0,0 +1,850 @@
+import           Cardano.CLI.Tx.BenchmarkingTxSubmission (ROEnv (..),
+                                                          TraceBenchTxSubmit (..),
+                                                          bulkSubmission)
+import           Cardano.CLI.Tx.Submission (submitTx)
+import           Cardano.CLI.Tx.BenchmarkingNodeToNode (BenchmarkTxSubmitTracers (..),
+                                                        SendRecvConnect,
+                                                        SendRecvTxSubmission,
+                                                        benchmarkConnectTxSubmit)
+import           Cardano.CLI.Tx.BenchmarkingTxSubClient
+
+import           Ouroboros.Consensus.Demo.Run (RunDemo)
+import           Ouroboros.Consensus.Node.Run (RunNode)
+import Ouroboros.Consensus.Block(BlockProtocol)
+import           Ouroboros.Consensus.Ledger.Byron.Config (ByronConfig,
+                                                          ByronEBBExtNodeConfig,
+                                                          pbftProtocolMagic)
+import           Ouroboros.Consensus.NodeId (CoreNodeId (..), NodeId (..))
+import           Ouroboros.Consensus.Node.ProtocolInfo (NumCoreNodes (..),
+                                                        ProtocolInfo (..),
+                                                        protocolInfo)
+import qualified Ouroboros.Consensus.Protocol as Consensus
+import           Ouroboros.Consensus.Ledger.Byron (ByronBlockOrEBB (..),
+                                                   GenTx (..))
+import qualified Ouroboros.Consensus.Ledger.Byron as Byron
+import           Ouroboros.Consensus.Protocol.Abstract (NodeConfig)
+import           Ouroboros.Consensus.Protocol.ExtNodeConfig (encNodeConfigExt)
+import           Ouroboros.Consensus.Protocol.WithEBBs (NodeConfig (WithEBBNodeConfig))
+
+newtype NumberOfTxs =
+  NumberOfTxs Word64
+  deriving (Eq, Ord, Show)
+
+newtype NumberOfOutputsPerTx =
+  NumberOfOutputsPerTx Int
+  deriving (Eq, Ord, Show)
+
+newtype FeePerTx =
+  FeePerTx Word64
+  deriving (Eq, Ord, Show)
+
+newtype TPSRate =
+  TPSRate Int
+  deriving (Eq, Ord, Show)
+
+-----------------------------------------------------------------------------------------
+-- | Genesis benchmark runner (we call it in 'Run.runNode').
+--
+--   Using a _richman_ (from genesis block) to supply some initial
+--   amount of funds for disbursment.
+-----------------------------------------------------------------------------------------
+genesisBenchmarkRunner
+  :: RunDemo (ByronBlockOrEBB ByronConfig)
+  => Tracer IO (TraceBenchTxSubmit (Byron.GenTxId (ByronBlockOrEBB ByronConfig)))
+  -> Tracer IO SendRecvConnect
+  -> Tracer IO (SendRecvTxSubmission (ByronBlockOrEBB ByronConfig))
+  -> Tracer IO String
+  -> CardanoConfiguration
+  -> Consensus.Protocol (ByronBlockOrEBB ByronConfig)
+  -> TopologyInfo
+  -> NumberOfTxs
+  -> NumberOfOutputsPerTx
+  -> FeePerTx
+  -> TPSRate
+  -> [FilePath]
+  -> IO ()
```

Comment:
I think many of the arguments should be grouped into a `data` type, like `BenchmarkConfig` or somesuch.
Same for the tracers, they should also benefit from grouping -- `data BencmarkTracers` maybe..

This would help with being able to just pass it down to `runBenchmark`.